test/e2e/middleware-custom-matchers-i18n/app/middleware.js

```diff
@@ -0,0 +1,19 @@
+import { NextResponse } from 'next/server'
+
+export default function middleware(request) {
+  const res = NextResponse.rewrite(new URL('/', request.url))
```

Comment:
Similar here using `new URL()` breaks automatic locale handling should use `request.nextUrl.pathname = '/'` and pass `request.nextUrl` instead.